src/newshop.cpp

```diff
@@ -503,6 +534,8 @@ bool install_ware_in_target_character(struct obj_data *ware, struct char_data *i
   // You must have the index to support it.
   else if (GET_OBJ_TYPE(ware) == ITEM_BIOWARE) {
     int esscost = GET_BIOWARE_ESSENCE_COST(ware);
+    if (GET_RACE(recipient) >= RACE_DRAKE_HUMAN && GET_RACE(recipient) <= RACE_DRAKE_TROLL)
+      esscost = GET_BIOWARE_ESSENCE_COST(ware) *= 2;
```

Comment:
This isn't quite right-- what you've written is "double the essence cost of the bioware, set the bioware itself to take that doubled essence cost, and then take the result of that as our new `esscost` value". This will probably(?) work, but you're changing the base value of the item itself, so when you go to uninstall it later and double it there as well, you'll add back too much bioware essence.

Change to `esscost *= 2;` so you double the existing essence cost without touching the esscost of the 'ware.

Reply:
395 is for x2 bioware cost, 537 is for x2 cyberware cost, drakes have double loss for both cyber/bio, or am I not seeing something?

Also, if I change it to esscost *= 2; it only returns half the bioindex on uninstall

Reply:
537 is for bioware cost because the 535 ifcheck is for `ITEM_BIOWARE`. 395 _should_ have no effect because `biocost` is not used outside of your scope, but you're doubling the actual bioware object's essence in the process.

The fact that it's only returning half the bioindex on uninstall means that you're missing code to double the bioindex returned on uninstall. Try restoring your previous code, installing bioware, and then doing a copyover before uninstalling it-- you'll only get half back there as well, as your change to the bioware cost is not persisted across copyovers.

Reply:
Ok, one last shot, see if that does the trick